src/hotspot/share/opto/loopopts.cpp

```diff
@@ -2223,10 +2212,6 @@ void PhaseIdealLoop::clone_loop_handle_data_uses(Node* old, Node_List &old_new,
         if( hit )             // Go ahead and re-hash for hits.
           _igvn.replace_node( use, hit );
       }
-
-      // If 'use' was in the loop-exit block, it now needs to be sunk
-      // below the post-loop merge point.
```

Comment:
This comment needs to be moved to the new lazy-replace logic and cloning needs to be explained there as well.